File: frontend/src/types/customer.ts
export interface Customer {
  id: number;
  name: string;
  email: string | null;
  phone: string | null;
  created_at: string;
  total_orders: number;
  total_spent: number;
  last_order_date: string | null;
}

export type CustomerCreate = {
  name: string;
  email: string | null;
  phone: string | null;
};

export type CustomerUpdate = {
  name?: string;
  email?: string | null;
  phone?: string | null;
};

File: backend/app/schemas/customer.py
from datetime import datetime

from pydantic import BaseModel, Field


class CustomerBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    email: str | None = Field(None, max_length=255)
    phone: str | None = Field(None, max_length=50)


class CustomerCreate(CustomerBase):
    pass


class CustomerUpdate(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=255)
    email: str | None = Field(None, max_length=255)
    phone: str | None = Field(None, max_length=50)


class CustomerResponse(CustomerBase):
    id: int
    created_at: datetime
    total_orders: int
    total_spent: float
    last_order_date: datetime | None
